snuba/subscriptions/worker.py

```diff
@@ -127,7 +127,12 @@ def flush_batch(
         # them to complete. Again, either the entire batch succeeds, or the
         # entire batch fails.
         for future in as_completed(
-            [self.__producer.produce(self.__topic, result) for result in results]
+            [
+                self.__producer.produce(
+                    self.__topic, subscription_task_result_codec.encode(result)
```

Comment:
See description re: coupling, there is also a more descriptive comment on the alternative approach in `snuba/utils/streams/synchronized.py` which would be useful here.